### DISCUSSION TITLE
Previous signal handler replaced with newly added

### DISCUSSION THREAD

**Discussion by sergiyha** [friflo/Friflo.Json.Fliox#46] [General]
The first signal lambda won't be invoked. Is it OK?

```cs
var store = new EntityStore();
var entity = store.CreateEntity();

entity.AddSignalHandler<MySignal>(signal => { Console.WriteLine($"s1: {signal.Event.I}"); });  
entity.AddSignalHandler<MySignal2>(signal => { Console.WriteLine($"s2: {signal.Event.I}"); });  
entity.EmitSignal(new MySignal { I = 1 });

entity.EmitSignal(new MySignal2 { I = 1 });


public struct MySignal
{
	public int I;
}

public struct MySignal2
{
	public int I;
}

```

**Comment by friflo** [comment]
Hi @sergiyha,

> The first signal lambda won't be invoked. Is it OK?

It must be invoked. Your expectation is correct. Its a bug.
Will look into this. I guess I can provide a fixed nuget package tomorrow.

**Comment by friflo** [comment]
Hi @sergiyha,

fixed the issue you discovered and published the nuget package `engine-v2.0.0-preview.4` including the fix.

Btw: The event & signal handlers added to an entity can be retrieved with
```csharp
var handlers = entity.DebugEventHandlers;
```


Thx, for reporting the issue!
